### ISSUE TITLE
Only add ASIN identifier if it is not the ISBN

### ISSUE THREAD

**Issue by hornc** [internetarchive/openlibrary#2312]

**Comment by cdrini** [comment]
I'm not sure of the context here, but I think it might better if the ASIN duplicates the ISBN. Otherwise the case where `asin=None` becomes ambiguous; does it mean the book doesn't have an ASIN? Or does it mean the ASIN is the ISBN?

**Comment by LeadSongDog** [comment]
@cdrini 
It's already ambiguous: asin=None could mean it has never been populated (so is undefined), or that amz never listed it (so it is nonexistent), or that there is an invalid ISBN that has promulgated (so it should not be used except to redirect). By far the most common situation is that an asin has been issued for a nonexistent book (usually based on prepublication data that did not pan out).

In any case, when it has the same value as the isbn, the asin adds no useful information: both will serve to access amz services.

**Comment by cdrini** [comment]
Yes, that's true, good catch! I'm thinking `ASIN=1234; ISBN=1234` is unambiguous, and `ASIN=None; ISBN=1234` is. The latter has 2 possible states: 1) ASIN exists but we don't know what it is, 2) this book does not have an ASIN. The former has only 1: This is the ASIN and this is the ISBN.

So when it has the same value as the isbn, the useful information it adds is _guaranteeing_ the existence of an Amazon record. It removes the ambiguity of the `ASIN=None; ISBN=1234` state.

To be mathematical about it, here are the possible cases if we use `ASIN=None` to imply `ASIN=ISBN`:
- Case 1 (`ASIN=None; ISBN=1234`):
  - `ASIN=not set, ISBN=1234`
  - `ASIN=nonexistent, ISBN=1234`
  - `ASIN=1234, ISBN=1234`
- Case 2 (`ASIN=1234; ISBN=1234`):
  - `ASIN=1234; ISBN=1234`

The number of possible states has decreased, so specifying the `ASIN` (even if it is the same as the `ISBN`) must have added information.

**Comment by LeadSongDog** [comment]
I suppose there could be valid ISBNs for which no corresponding ASIN would exist: such might arise for competing retailers' private-label books. Similarly, valid ASINs exist for many books without ISBNs, particularly kindle ebooks and old (pre-ISBN) used books. Likewise, invalid ASINs exist, as do invalid ISBNs. However, so long as amz responds to queries on an ASIN we have no way of distinguishing between ASINs for books which are unavailable and those which never existed. We would essentially have to just trust them, while fully understanding that they can't be trusted.

**Comment by cdrini** [comment]
> However, so long as amz responds to queries on an ASIN we have no way of distinguishing between ASINs for books which are unavailable and those which never existed. We would essentially have to just trust them, while fully understanding that they can't be trusted.

Yep! :P Unfortunately I don't think there's a way around that when dealing with a private company's identifiers. :/

**Comment by LeadSongDog** [comment]
I suppose we could start by ignoring ASINs that have nothing but the most basic information.  If a "currently unavailable" ASIN is not an ISBN and shows an author that cannot be matched elsewhere to the title, it is usually a bogus book, or at least the metadata is crap that we should not be propagating.  For instance, B009UDUY86 on the surface looks to be just another print-on-demand version of a PD-1923 book, but Pierre Albert Monier was only born in 1905, and Worldcat shows the earliest edition of _The complete technique of making films_ as 1958 and his earliest work as the 1948 _Le livre du cinéaste amateur; technique, pratique, esthétique_.
The point is that a private identifier's metadata should match at least two of author, title, publisher *AND* be shown as available. Without some reasonable corroboration, we should ignore them.

**Comment by hornc** [comment]
@cdrini I added the feature to add ASINs when imported recently #2218 and since then the value of these imports, and ASINs as a reliable identifier, have been brought into question, see #2674 

Highlighting the ISBN10 as an ASIN seems now to be merely duplicating data, and the provenance, where is is relevant, is already captured by the `"source_records": "amazon:<id>"

Adding this identifier: amazon for an ISBN is effectively adding a _third_ copy of the ISBN.

There have been other issues closed where we have said we are not going to add new fields for booksellers identifiers which simply reuse ISBNs as IDs. I think ASIN were originally felt to be different, but #2674 indicates that ASIN are only valuable _when_ they are ISBNs, and otherwise they are merely seller inventory numbers that can't be relied upon to exist, or point to a single book.

None of our existing Amazon affiliate links rely on this field being populated -- the code uses ISBN first and falls back to this id if there is no ISBN.  #2674 suggests we perhaps shouldn't even do the fall back as it is not guaranteed to be reliable.

I don't think this removes any functionality, it simply reduces data duplication that was added with no positive effect in #2218
